core/modules/cart/store/actions.ts

```diff
@@ -482,78 +380,54 @@ const actions: ActionTree<CartState, RootState> = {
     Logger.warn('The "cart/refreshTotals" action is deprecated and will not be supported with the Vue Storefront 1.11', 'cart')()
     return dispatch('syncTotals', payload)
   },
-  /** remove discount code from the cart + sync totals @description this method is part of "public" cart API */
   async removeCoupon ({ getters, dispatch }) {
     if (getters.isTotalsSyncEnabled && getters.isCartConnected) {
-      const task = await TaskQueue.execute({ url: config.cart.deletecoupon_endpoint,
-        payload: {
-          method: 'POST',
-          headers: { 'Content-Type': 'application/json' },
-          mode: 'cors'
-        },
-        silent: false
-      })
-      if (task.result) {
+      const { result } = await CartService.removeCoupon()
+
+      if (result) {
         dispatch('syncTotals', { forceServerSync: true })
-        return task.result
+        return result
```

Comment:
Are you sure about this change? The motivation is to align all actions to always return task DTO? Please make sure that the theme isn’t using these direct results

We really must keep up to this backward compatibility seriously otherwise partners get pissed off :)

Reply:
I've checked, previously both `removeCupon` and `applyCoupon` returned just `result`